plugins/chat/spec/jobs/scheduled/auto_join_users_spec.rb

```diff
@@ -3,8 +3,18 @@
 describe Jobs::Chat::AutoJoinUsers do
   subject(:job) { described_class.new }
 
+  before { Jobs.run_immediately! }
+
   it "works" do
-    Jobs.run_immediately!
+    _staged_user = Fabricate(:user, staged: true)
+    _suspended_user = Fabricate(:user, suspended_till: 1.day.from_now)
+    _inactive_user = Fabricate(:user, active: false)
```

Comment:
For completeness sake, I think we need the following users too:

1. User with `UserOption#chat_enabled` that is `false`
2. User that has an `AnonymousUser` record

Reply:
will add